cmd/serverless/main.go

```diff
@@ -70,32 +71,28 @@ const (
 )
 
 func main() {
-	flavor.SetFlavor(flavor.ServerlessAgent)
-	config.Datadog.Set("use_v2_api.series", false, model.SourceAgentRuntime)
-	stopCh := make(chan struct{})
-
-	// Disable remote configuration for now as it just spams the debug logs
-	// and provides no value.
-	os.Setenv("DD_REMOTE_CONFIGURATION_ENABLED", "false")
-
 	// run the agent
-	serverlessDaemon, err := runAgent(stopCh)
+	err := fxutil.OneShot(runAgent)
+
 	if err != nil {
 		log.Error(err)
 		os.Exit(-1)
 	}
-
-	// handle SIGTERM signal
-	go handleSignals(serverlessDaemon, stopCh)
-
-	// block here until we receive a stop signal
-	<-stopCh
 }
 
-func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error) {
+func runAgent() (err error) {
```

Comment:
It seems this function never returns a non nil error.
I would recommend to either:
- remove the error from the signature
- return a real error and let the outside calling function logging it, instead of
```golang
log.Errorf("Can't register as a serverless agent: %s", err)
		return
```

Reply:
Agreed that this is worthwhile improvement. I went with the approach of returning an error and letting the outside calling function log it

Reply:
After further review I think it's nest to go with the option of removing the error from the signature. Like you mentioned the `runAgent` function in its current state never returns a non nil error, so the below code is effectively unreachable. In order to maintain the current behavior I removed this block of code and left the error logs as is. Let me know if it looks good to you!

https://github.com/DataDog/datadog-agent/blob/8c0f4fa0003823d45153a959551e580977140516/cmd/serverless/main.go#L83-L86